### PR TITLE
update docs about global kill

### PR DESCRIPTION
**Pull request by xuyifangreeneyes**
### What is changed, added or deleted? (Required)

Update docs about global kill.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/issues/8854

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@xuyifangreeneyes 能请 @ 一两个技术 review 吗？谢谢～

File: sql-statements/sql-statement-kill.md [hunk 1]
```diff
@@ -49,6 +49,8 @@ Query OK, 0 rows affected (0.00 sec)
 * 按照设计，`KILL` 语句默认与 MySQL 不兼容。负载均衡器后面通常放有多个 TiDB 服务器，这种默认不兼容有助于防止在错误的 TiDB 服务器上终止连接。
 * 请**不要**在配置文件里设置 [`compatible-kill-query = true`](/tidb-configuration-file.md#compatible-kill-query)，**除非**你确定客户端将始终连接到同一个 TiDB 节点。这是因为当你在默认的 MySQL 客户端按下 <kbd>ctrl</kbd>+<kbd>c</kbd> 时，客户端会开启一个新连接，并在这个新连接中执行 `KILL` 语句。此时，如果客户端和 TiDB 中间有代理，新连接可能会被路由到其他的 TiDB 节点，从而错误地终止其他会话。
 * `KILL TIDB` 语句是 TiDB 的扩展语法，其功能与 MySQL 命令 `KILL [CONNECTION|QUERY]` 和 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 相同。在同一个 TiDB 节点上，你可以安全地使用 `KILL TIDB` 语句。
+* 从 TiDB v6.1.0 起，`enable-global-kill` 默认值为 true。当 `enable-global-kill` 值为 true 时，无论`KILL` 语句和 `KILL TIDB` 语句均能跨实例终止查询或连接，当客户端和 TiDB 中间有代理时，`KILL` 语句和 `KILL TIDB` 语句会被转发给对应的 TiDB 实例执行，无需担心错误地终止其他查询或连接。
```
Comment (ran-huang) [inline sql-statements/sql-statement-kill.md:52]: `enable-global-kill` 是新增的 config 还是已有的？在现有的配置文件里找不到这个

Comment (xuyifangreeneyes) [reply sql-statements/sql-statement-kill.md]: 是新增的 config。刚刚更新文档添加了这个 config。

File: tidb-configuration-file.md [hunk 2]
```diff
@@ -170,12 +170,18 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
 + 当该配置项值为 `true` 时，`ENUM` 和 `SET` 单个元素的最大长度为 255 个字符，[与 MySQL 8 兼容](https://dev.mysql.com/doc/refman/8.0/en/string-type-syntax.html)；当该配置项值为 `false` 时，不对单个元素的长度进行限制，与 TiDB v5.0 之前的版本兼容。
 
-#### `graceful-wait-before-shutdown` <span class="version-mark">从 v5.0 版本开始引入</span>
+### `graceful-wait-before-shutdown` <span class="version-mark">从 v5.0 版本开始引入</span>
 
 - 指定关闭服务器时 TiDB 等待的秒数，使得客户端有时间断开连接。
 - 默认值：0
 - 在 TiDB 等待服务器关闭期间，HTTP 状态会显示失败，使得负载均衡器可以重新路由流量。
 
+### `enable-global-kill` <span class="version-mark">从 v5.0 版本开始引入</span>
```
Comment (ran-huang) [inline tidb-configuration-file.md:179]:
```suggestion
### `enable-global-kill` <span class="version-mark">从 v6.0 版本开始引入</span>
```

File: tidb-configuration-file.md [hunk 3]
```diff
@@ -170,12 +170,18 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
 
 - 指定关闭服务器时 TiDB 等待的秒数，使得客户端有时间断开连接。
 - 默认值：0
 - 在 TiDB 等待服务器关闭期间，HTTP 状态会显示失败，使得负载均衡器可以重新路由流量。
 
+### `enable-global-kill` <span class="version-mark">从 v6.1.0 版本开始引入</span>
+
++ 用于开启 global kill 功能。
++ 默认值：true
++ 当该配置项值为 `true` 时，`KILL` 语句和 `KILL TIDB` 语句均能跨节点终止查询或连接，无需担心错误地终止其他查询或连接。用户使用客户端连接到任何一个 TiDB 节点执行 `KILL` 语句或 `KILL TIDB` 语句，该语句会被转发给对应的 TiDB 节点。当客户端和 TiDB 中间有代理时，`KILL` 语句或 `KILL TIDB` 语句也会被转发给对应的 TiDB 节点执行。目前暂时不支持在`enable-global-kill` 为 `true` 时用 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 终止查询或连接。请参考 [KILL [TIDB]](/sql-statements/sql-statement-kill.md)。
```
Comment (qiancai) [inline tidb-configuration-file.md:183]:
```suggestion
+ 当该配置项值为 `true` 时，`KILL` 语句和 `KILL TIDB` 语句均能跨节点终止查询或连接，无需担心错误地终止其他查询或连接。当你使用客户端连接到任何一个 TiDB 节点执行 `KILL` 语句或 `KILL TIDB` 语句，该语句会被转发给对应的 TiDB 节点。当客户端和 TiDB 中间有代理时，`KILL` 语句或 `KILL TIDB` 语句也会被转发给对应的 TiDB 节点执行。目前暂时不支持在 `enable-global-kill` 为 `true` 时用 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 终止查询或连接。关于 `KILL` 语句的更多信息，请参考 [KILL [TIDB]](/sql-statements/sql-statement-kill.md)。
```

File: sql-statements/sql-statement-kill.md [hunk 4]
```diff
@@ -49,6 +49,7 @@ Query OK, 0 rows affected (0.00 sec)
 * 按照设计，`KILL` 语句默认与 MySQL 不兼容。负载均衡器后面通常放有多个 TiDB 服务器，这种默认不兼容有助于防止在错误的 TiDB 服务器上终止连接。
 * 请**不要**在配置文件里设置 [`compatible-kill-query = true`](/tidb-configuration-file.md#compatible-kill-query)，**除非**你确定客户端将始终连接到同一个 TiDB 节点。这是因为当你在默认的 MySQL 客户端按下 <kbd>ctrl</kbd>+<kbd>c</kbd> 时，客户端会开启一个新连接，并在这个新连接中执行 `KILL` 语句。此时，如果客户端和 TiDB 中间有代理，新连接可能会被路由到其他的 TiDB 节点，从而错误地终止其他会话。
 * `KILL TIDB` 语句是 TiDB 的扩展语法，其功能与 MySQL 命令 `KILL [CONNECTION|QUERY]` 和 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 相同。在同一个 TiDB 节点上，你可以安全地使用 `KILL TIDB` 语句。
+* TiDB v6.1 引入了配置项 [`enable-global-kill`](/tidb-configuration-file.md#enable-global-kill)，默认值为 `true`。当 `enable-global-kill` 为 `true` 时，`KILL` 语句和 `KILL TIDB` 语句均能跨节点终止查询或连接，无需担心错误地终止其他查询或连接。用户使用客户端连接到任何一个 TiDB 节点执行 `KILL` 语句或 `KILL TIDB` 语句，该语句会被转发给对应的 TiDB 节点。当客户端和 TiDB 中间有代理时，`KILL` 语句或 `KILL TIDB` 语句也会被转发给对应的 TiDB 节点执行。目前暂时不支持在`enable-global-kill` 为 `true` 时用 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 终止查询或连接。
```
Comment (qiancai) [inline sql-statements/sql-statement-kill.md:52]:
```suggestion
* TiDB 从 v6.1.0 起引入 [`enable-global-kill`](/tidb-configuration-file.md#enable-global-kill) 配置项，默认值为 `true`。当 `enable-global-kill` 为 `true` 时，`KILL` 语句和 `KILL TIDB` 语句均能跨节点终止查询或连接，无需担心错误地终止其他查询或连接。当你使用客户端连接到任何一个 TiDB 节点执行 `KILL` 语句或 `KILL TIDB` 语句，该语句会被转发给对应的 TiDB 节点。当客户端和 TiDB 中间有代理时，`KILL` 语句或 `KILL TIDB` 语句也会被转发给对应的 TiDB 节点执行。目前暂时不支持在 `enable-global-kill` 为 `true` 时用 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 终止查询或连接。
```

File: sql-statements/sql-statement-kill.md [hunk 5]
```diff
@@ -49,6 +49,7 @@ Query OK, 0 rows affected (0.00 sec)
 * 按照设计，`KILL` 语句默认与 MySQL 不兼容。负载均衡器后面通常放有多个 TiDB 服务器，这种默认不兼容有助于防止在错误的 TiDB 服务器上终止连接。
 * 请**不要**在配置文件里设置 [`compatible-kill-query = true`](/tidb-configuration-file.md#compatible-kill-query)，**除非**你确定客户端将始终连接到同一个 TiDB 节点。这是因为当你在默认的 MySQL 客户端按下 <kbd>ctrl</kbd>+<kbd>c</kbd> 时，客户端会开启一个新连接，并在这个新连接中执行 `KILL` 语句。此时，如果客户端和 TiDB 中间有代理，新连接可能会被路由到其他的 TiDB 节点，从而错误地终止其他会话。
 * `KILL TIDB` 语句是 TiDB 的扩展语法，其功能与 MySQL 命令 `KILL [CONNECTION|QUERY]` 和 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 相同。在同一个 TiDB 节点上，你可以安全地使用 `KILL TIDB` 语句。
+* TiDB 从 v6.1.0 起引入 [`enable-global-kill`](/tidb-configuration-file.md#enable-global-kill) 配置项，默认值为 `true`。当 `enable-global-kill` 为 `true` 时，`KILL` 语句和 `KILL TIDB` 语句均能跨节点终止查询或连接，无需担心错误地终止其他查询或连接。当你使用客户端连接到任何一个 TiDB 节点执行 `KILL` 语句或 `KILL TIDB` 语句，该语句会被转发给对应的 TiDB 节点。当客户端和 TiDB 中间有代理时，`KILL` 语句或 `KILL TIDB` 语句也会被转发给对应的 TiDB 节点执行。目前暂时不支持在 `enable-global-kill` 为 `true` 时用 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 终止查询或连接。
```
Comment (qiancai) [inline sql-statements/sql-statement-kill.md:52]:
```suggestion
* TiDB 从 v6.1.0 起引入 [`enable-global-kill`](/tidb-configuration-file.md#enable-global-kill) 配置项，默认值为 `true`。当 `enable-global-kill` 为 `true` 时，`KILL` 语句和 `KILL TIDB` 语句均能跨节点终止查询或连接，无需担心错误地终止其他查询或连接。当你使用客户端连接到任何一个 TiDB 节点执行 `KILL` 语句或 `KILL TIDB` 语句时，该语句会被转发给对应的 TiDB 节点。当客户端和 TiDB 中间有代理时，`KILL` 语句或 `KILL TIDB` 语句也会被转发给对应的 TiDB 节点执行。目前暂时不支持在 `enable-global-kill` 为 `true` 时用 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 终止查询或连接。
```

File: tidb-configuration-file.md [hunk 6]
```diff
@@ -170,12 +170,18 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
-#### `graceful-wait-before-shutdown` <span class="version-mark">从 v5.0 版本开始引入</span>
+### `graceful-wait-before-shutdown` <span class="version-mark">从 v5.0 版本开始引入</span>
 
 - 指定关闭服务器时 TiDB 等待的秒数，使得客户端有时间断开连接。
 - 默认值：0
 - 在 TiDB 等待服务器关闭期间，HTTP 状态会显示失败，使得负载均衡器可以重新路由流量。
 
+### `enable-global-kill` <span class="version-mark">从 v6.1.0 版本开始引入</span>
+
++ 用于开启 global kill 功能。
```
Comment (qiancai) [inline tidb-configuration-file.md:181]:
```suggestion
+ 用于开启 global kill（跨节点终止查询或连接）功能。
```

File: tidb-configuration-file.md [hunk 7]
```diff
@@ -170,12 +170,18 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
++ 默认值：true
++ 当该配置项值为 `true` 时，`KILL` 语句和 `KILL TIDB` 语句均能跨节点终止查询或连接，无需担心错误地终止其他查询或连接。当你使用客户端连接到任何一个 TiDB 节点执行 `KILL` 语句或 `KILL TIDB` 语句，该语句会被转发给对应的 TiDB 节点。当客户端和 TiDB 中间有代理时，`KILL` 语句或 `KILL TIDB` 语句也会被转发给对应的 TiDB 节点执行。目前暂时不支持在 `enable-global-kill` 为 `true` 时用 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 终止查询或连接。关于 `KILL` 语句的更多信息，请参考 [KILL [TIDB]](/sql-statements/sql-statement-kill.md)。
```
Comment (qiancai) [inline tidb-configuration-file.md:183]:
```suggestion
+ 当该配置项值为 `true` 时，`KILL` 语句和 `KILL TIDB` 语句均能跨节点终止查询或连接，无需担心错误地终止其他查询或连接。当你使用客户端连接到任何一个 TiDB 节点执行 `KILL` 语句或 `KILL TIDB` 语句时，该语句会被转发给对应的 TiDB 节点。当客户端和 TiDB 中间有代理时，`KILL` 语句或 `KILL TIDB` 语句也会被转发给对应的 TiDB 节点执行。目前暂时不支持在 `enable-global-kill` 为 `true` 时用 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 终止查询或连接。关于 `KILL` 语句的更多信息，请参考 [KILL [TIDB]](/sql-statements/sql-statement-kill.md)。
```

File: sql-statements/sql-statement-kill.md [hunk 8]
```diff
@@ -49,6 +49,7 @@ Query OK, 0 rows affected (0.00 sec)
 * 按照设计，`KILL` 语句默认与 MySQL 不兼容。负载均衡器后面通常放有多个 TiDB 服务器，这种默认不兼容有助于防止在错误的 TiDB 服务器上终止连接。
 * 请**不要**在配置文件里设置 [`compatible-kill-query = true`](/tidb-configuration-file.md#compatible-kill-query)，**除非**你确定客户端将始终连接到同一个 TiDB 节点。这是因为当你在默认的 MySQL 客户端按下 <kbd>ctrl</kbd>+<kbd>c</kbd> 时，客户端会开启一个新连接，并在这个新连接中执行 `KILL` 语句。此时，如果客户端和 TiDB 中间有代理，新连接可能会被路由到其他的 TiDB 节点，从而错误地终止其他会话。
 * `KILL TIDB` 语句是 TiDB 的扩展语法，其功能与 MySQL 命令 `KILL [CONNECTION|QUERY]` 和 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 相同。在同一个 TiDB 节点上，你可以安全地使用 `KILL TIDB` 语句。
+* TiDB 从 v6.1.0 起引入 [`enable-global-kill`](/tidb-configuration-file.md#enable-global-kill) 配置项，默认值为 `true`。当 `enable-global-kill` 为 `true` 时，`KILL` 语句和 `KILL TIDB` 语句均能跨节点终止查询或连接，无需担心错误地终止其他查询或连接。当你使用客户端连接到任何一个 TiDB 节点执行 `KILL` 语句或 `KILL TIDB` 语句时，该语句会被转发给对应的 TiDB 节点。当客户端和 TiDB 中间有代理时，`KILL` 语句或 `KILL TIDB` 语句也会被转发给对应的 TiDB 节点执行。目前暂时不支持在 `enable-global-kill` 为 `true` 时用 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 终止查询或连接。
```
Comment (qiancai) [inline sql-statements/sql-statement-kill.md:52]:
```suggestion
* TiDB 从 v6.1.0 起引入 [`enable-global-kill`](/tidb-configuration-file.md#enable-global-kill-从-v610-版本开始引入) 配置项，默认值为 `true`。当 `enable-global-kill` 为 `true` 时，`KILL` 语句和 `KILL TIDB` 语句均能跨节点终止查询或连接，无需担心错误地终止其他查询或连接。当你使用客户端连接到任何一个 TiDB 节点执行 `KILL` 语句或 `KILL TIDB` 语句时，该语句会被转发给对应的 TiDB 节点。当客户端和 TiDB 中间有代理时，`KILL` 语句或 `KILL TIDB` 语句也会被转发给对应的 TiDB 节点执行。目前暂时不支持在 `enable-global-kill` 为 `true` 时用 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 终止查询或连接。
```

File: tidb-configuration-file.md [hunk 9]
```diff
@@ -170,12 +170,18 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
 
 - 指定关闭服务器时 TiDB 等待的秒数，使得客户端有时间断开连接。
 - 默认值：0
 - 在 TiDB 等待服务器关闭期间，HTTP 状态会显示失败，使得负载均衡器可以重新路由流量。
 
+### `enable-global-kill` <span class="version-mark">从 v6.1.0 版本开始引入</span>
+
++ 用于开启 global kill（跨节点终止查询或连接）功能。
++ 默认值：true
++ 当该配置项值为 `true` 时，`KILL` 语句和 `KILL TIDB` 语句均能跨节点终止查询或连接，无需担心错误地终止其他查询或连接。当你使用客户端连接到任何一个 TiDB 节点执行 `KILL` 语句或 `KILL TIDB` 语句时，该语句会被转发给对应的 TiDB 节点。当客户端和 TiDB 中间有代理时，`KILL` 语句或 `KILL TIDB` 语句也会被转发给对应的 TiDB 节点执行。目前暂时不支持在 `enable-global-kill` 为 `true` 时用 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 终止查询或连接。关于 `KILL` 语句的更多信息，请参考 [KILL [TIDB]](/sql-statements/sql-statement-kill.md)。
```
Comment (chrysan) [inline tidb-configuration-file.md:183]: 是否可以减少一些重复描述，既然会 Link。

File: sql-statements/sql-statement-kill.md [hunk 10]
```diff
@@ -49,6 +49,7 @@ Query OK, 0 rows affected (0.00 sec)
 * 按照设计，`KILL` 语句默认与 MySQL 不兼容。负载均衡器后面通常放有多个 TiDB 服务器，这种默认不兼容有助于防止在错误的 TiDB 服务器上终止连接。
 * 请**不要**在配置文件里设置 [`compatible-kill-query = true`](/tidb-configuration-file.md#compatible-kill-query)，**除非**你确定客户端将始终连接到同一个 TiDB 节点。这是因为当你在默认的 MySQL 客户端按下 <kbd>ctrl</kbd>+<kbd>c</kbd> 时，客户端会开启一个新连接，并在这个新连接中执行 `KILL` 语句。此时，如果客户端和 TiDB 中间有代理，新连接可能会被路由到其他的 TiDB 节点，从而错误地终止其他会话。
 * `KILL TIDB` 语句是 TiDB 的扩展语法，其功能与 MySQL 命令 `KILL [CONNECTION|QUERY]` 和 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 相同。在同一个 TiDB 节点上，你可以安全地使用 `KILL TIDB` 语句。
+* TiDB 从 v6.1.0 起引入 [`enable-global-kill`](/tidb-configuration-file.md#enable-global-kill-从-v610-版本开始引入) 配置项，默认值为 `true`。当 `enable-global-kill` 为 `true` 时，`KILL` 语句和 `KILL TIDB` 语句均能跨节点终止查询或连接，无需担心错误地终止其他查询或连接。当你使用客户端连接到任何一个 TiDB 节点执行 `KILL` 语句或 `KILL TIDB` 语句时，该语句会被转发给对应的 TiDB 节点。当客户端和 TiDB 中间有代理时，`KILL` 语句或 `KILL TIDB` 语句也会被转发给对应的 TiDB 节点执行。目前暂时不支持在 `enable-global-kill` 为 `true` 时用 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 终止查询或连接。
```
Comment (chrysan) [inline sql-statements/sql-statement-kill.md:52]: 是否需要补充一下和 compatible-kill-query 的关系？

Comment (breezewish) [inline sql-statements/sql-statement-kill.md:52]: L49~51 也许可以考虑修改一下，因为 enable-global-kill 默认值是 true，即将对用户基本无感知？

Comment (breezewish) [reply sql-statements/sql-statement-kill.md:52]: 按照用户默认不感知 KILL TIDB 语法稍微修改了一下，可以看看是否合适：

`````markdown

---
title: KILL
summary: TiDB 数据库中 KILL 的使用概况。
aliases: ['/docs-cn/dev/sql-statements/sql-statement-kill/','/docs-cn/dev/reference/sql/statements/kill/']
---

# KILL

`KILL` 语句可以终止当前 TiDB 集群中任意一个 TiDB 实例中的某个连接。

## 语法图

```ebnf+diagram
KillStmt ::= 'KILL' 'TIDB'? ( 'CONNECTION' | 'QUERY' )? CONNECTION_ID
```

## 示例

### 查询当前集群中所有所有活跃查询，并终止其中某一个连接

{{< copyable "sql" >}}

```sql
SELECT ID, USER, INSTANCE, INFO FROM INFORMATION_SCHEMA.CLUSTER_PROCESSLIST;
```

```
+---------------------+------+-----------------+-----------------------------------------------------------------------------+
| ID                  | USER | INSTANCE        | INFO                                                                        |
+---------------------+------+-----------------+-----------------------------------------------------------------------------+
| 8306449708033769879 | root | 127.0.0.1:10082 | select sleep(30), 'foo'                                                     |
| 5857102839209263511 | root | 127.0.0.1:10080 | select sleep(50)                                                            |
| 5857102839209263513 | root | 127.0.0.1:10080 | SELECT ID, USER, INSTANCE, INFO FROM INFORMATION_SCHEMA.CLUSTER_PROCESSLIST |
+---------------------+------+-----------------+-----------------------------------------------------------------------------+
```

{{< copyable "sql" >}}

```sql
KILL 5857102839209263511;
```

```
Query OK, 0 rows affected (0.00 sec)
```

## MySQL 兼容性

- MySQL 的 `KILL` 语句仅能终止当前连接的 MySQL 实例上的连接，TiDB 的 `KILL` 语句能终止整个集群中任意一个 TiDB 实例上的连接。
- 暂时不支持使用 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 终止查询或连接。

## 行为变更说明

TiDB 从 v6.1.0 起新增 Global Kill 功能（由 [`enable-global-kill`](/tidb-configuration-file.md#enable-global-kill-从-v610-版本开始引入) 配置项控制，默认启用）。启用 Global Kill 功能时，`KILL` 语句和 `KILL TIDB` 语句均能跨节点终止查询或连接，且无需担心错误地终止其他查询或连接。当你使用客户端连接到任何一个 TiDB 节点执行 `KILL` 语句或 `KILL TIDB` 语句时，该语句会被转发给对应的 TiDB 节点。当客户端和 TiDB 中间有代理时，`KILL` 及 `KILL TIDB` 语句也会被转发给对应的 TiDB 节点执行。

对于 TiDB v6.1.0 之前的版本，或未启用 Global Kill 功能时：

- `KILL` 语句默认与 MySQL 不兼容，负载均衡器后面通常放有多个 TiDB 服务器，这种默认不兼容有助于防止在错误的 TiDB 服务器上终止连接。你需要显式地增加 `TIDB` 后缀，通过执行 `KILL TIDB` 语句来终止当前连接的 TiDB 实例上的其他连接。
- **强烈不建议**在配置文件里设置 [`compatible-kill-query = true`](/tidb-configuration-file.md#compatible-kill-query)，**除非**你确定客户端将始终连接到同一个 TiDB 节点。这是因为当你在默认的 MySQL 客户端按下 <kbd>ctrl</kbd>+<kbd>c</kbd> 时，客户端会开启一个新连接，并在这个新连接中执行 `KILL` 语句。此时，如果客户端和 TiDB 中间有代理，新连接可能会被路由到其他的 TiDB 节点，从而错误地终止其他会话。
- `KILL TIDB` 语句是 TiDB 的扩展语法，其功能与 MySQL 命令 `KILL [CONNECTION|QUERY]` 和 MySQL 命令行 <kbd>ctrl</kbd>+<kbd>c</kbd> 相同。在同一个 TiDB 节点上，你可以安全地使用 `KILL TIDB` 语句。

## 另请参阅

- [SHOW \[FULL\] PROCESSLIST](/sql-statements/sql-statement-show-processlist.md)
- [CLUSTER_PROCESSLIST](/information-schema/information-schema-processlist.md#cluster_processlist)

`````

File: tidb-configuration-file.md (hunk 9 above)
Comment (qiancai) [reply tidb-configuration-file.md:183]: 感觉不减少也行，这样如果看明白了，也不需要再去打开 link 看。

File: sql-statements/sql-statement-kill.md (hunk 10 above)
Comment (xuyifangreeneyes) [reply sql-statements/sql-statement-kill.md:52]: `enble-global-kill` 为 true 的时候，`compatible-kill-query` 无论 true 还是 false 都不影响 kill 行为了

File: tidb-configuration-file.md [hunk 11]
```diff
@@ -170,12 +170,18 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
-#### `graceful-wait-before-shutdown` <span class="version-mark">从 v5.0 版本开始引入</span>
+### `graceful-wait-before-shutdown` <span class="version-mark">从 v5.0 版本开始引入</span>
 
 - 指定关闭服务器时 TiDB 等待的秒数，使得客户端有时间断开连接。
 - 默认值：0
 - 在 TiDB 等待服务器关闭期间，HTTP 状态会显示失败，使得负载均衡器可以重新路由流量。
 
+### `enable-global-kill` <span class="version-mark">从 v6.1.0 版本开始引入</span>
+
++ 用于开启 global kill（跨节点终止查询或连接）功能。
```
Comment (qiancai) [inline tidb-configuration-file.md:181]:
```suggestion
+ 用于开启 Global Kill（跨节点终止查询或连接）功能。
```

File: sql-statements/sql-statement-kill.md [hunk 12]
```diff
@@ -1,43 +1,43 @@
...
 ```ebnf+diagram
-KillStmt ::= KillOrKillTiDB ( 'CONNECTION' | 'QUERY' )? NUM
-
-KillOrKillTiDB ::= 'KILL' 'TIDB'?
+KillStmt ::= 'KILL' 'TIDB'? ( 'CONNECTION' | 'QUERY' )? CONNECTION_ID
 ```
 
 ## 示例
 
+### 查询当前集群中所有所有活跃查询，并终止其中某一个连接
```
Comment (qiancai) [inline sql-statements/sql-statement-kill.md:19]:
```suggestion
查询当前集群中所有所有活跃查询，并终止其中某一个连接：
```

**Review by chrysan — APPROVED** [review]

**Review by breezewish — APPROVED** [review]

**Review by shichun-0415 — APPROVED** [review]
LGTM

**Review by qiancai — APPROVED** [review]

**Pull request merged**